src/shims/foreign_items/posix/linux.rs

```diff
@@ -143,12 +157,13 @@ fn getrandom<'tcx>(
     args: &[OpTy<'tcx, Tag>],
     dest: PlaceTy<'tcx, Tag>,
 ) -> InterpResult<'tcx> {
-    let ptr = this.read_scalar(args[0])?.not_undef()?;
-    let len = this.read_scalar(args[1])?.to_machine_usize(this)?;
+    let &[ptr, len, _flags] = check_args(args)?;
```

Comment:
This will, however, show a wrong error... namely if you pass 3 arguments it will show "passed 2 arguments expected 3", whereas really it was "passed 3 arguments expected 4".

Probably it is better to change the type signature here to just take 3 arguments fixed, and let the caller do `check_args`.